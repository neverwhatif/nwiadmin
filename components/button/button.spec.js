import React from 'react';
import { shallow } from 'enzyme';

import Button, { PrimaryButton } from './index';

jest.mock('nwiadmin/components/link', () => 'Link');
const mockOnClick = jest.fn();

describe('Button', () => {
    const component = shallow(<Button>Lorem</Button>);
    const componentWithStyle = shallow(<Button buttonStyle="lorem">Lorem</Button>);
    const componentDisabled = shallow(<Button isDisabled>Lorem</Button>);
    const componentLink = shallow(<Button to="lorem">Lorem</Button>);
    const componentOnClick = shallow(<Button onClick={() => mockOnClick()}>Lorem</Button>);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentWithStyle).toMatchSnapshot();
        expect(componentDisabled).toMatchSnapshot();
        expect(componentLink).toMatchSnapshot();
        expect(componentOnClick).toMatchSnapshot();
    });
    it('should render text', () => {
        expect(component.text()).toBe('Lorem');
    });
    it('should render styles', () => {
        expect(componentWithStyle.hasClass('rootLorem')).toBe(true);
    });
    it('should have a default onClick', () => {
        expect(Button.defaultProps.onClick()).toBe(null);
    });
    it('should be disabled', () => {
        expect(componentDisabled.prop('disabled')).toBe(true);
    });
    it('should render link', () => {
        expect(componentLink.find('Link').length).toBe(1);
        expect(componentLink.prop('to')).toBe('lorem');
    });
    it('should handle click', () => {
        componentOnClick.simulate('click');
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
});

describe('PrimaryButton', () => {
    const component = shallow(<PrimaryButton>Lorem</PrimaryButton>);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render a button', () => {
        expect(component.children().length).toBe(1);
        expect(component.containsMatchingElement(<Button>Lorem</Button>)).toBe(true);
    });
    it('should render primary styles', () => {
        expect(component.dive().hasClass('rootPrimary')).toBe(true);
    });
});
