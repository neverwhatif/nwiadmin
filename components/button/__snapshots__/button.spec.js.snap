// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    buttonStyle={null}
    isDisabled={false}
    onClick={[Function]}
    style={null}
    to={null}
  >
    Lorem
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "root",
      "disabled": false,
      "onClick": [Function],
      "style": null,
      "type": "button",
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "root",
        "disabled": false,
        "onClick": [Function],
        "style": null,
        "type": "button",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    buttonStyle="lorem"
    isDisabled={false}
    onClick={[Function]}
    style={null}
    to={null}
  >
    Lorem
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "root rootLorem",
      "disabled": false,
      "onClick": [Function],
      "style": null,
      "type": "button",
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "root rootLorem",
        "disabled": false,
        "onClick": [Function],
        "style": null,
        "type": "button",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button should render 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    buttonStyle={null}
    isDisabled={true}
    onClick={[Function]}
    style={null}
    to={null}
  >
    Lorem
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "root rootDisabled",
      "disabled": true,
      "onClick": [Function],
      "style": null,
      "type": "button",
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "root rootDisabled",
        "disabled": true,
        "onClick": [Function],
        "style": null,
        "type": "button",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button should render 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    buttonStyle={null}
    isDisabled={false}
    onClick={[Function]}
    style={null}
    to="lorem"
  >
    Lorem
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "root",
      "to": "lorem",
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "Link",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "root",
        "to": "lorem",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "Link",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button should render 5`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    buttonStyle={null}
    isDisabled={false}
    onClick={[Function]}
    style={null}
    to={null}
  >
    Lorem
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "root",
      "disabled": false,
      "onClick": [Function],
      "style": null,
      "type": "button",
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "root",
        "disabled": false,
        "onClick": [Function],
        "style": null,
        "type": "button",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PrimaryButton should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrimaryButton>
    Lorem
  </PrimaryButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "buttonStyle": "primary",
      "children": "Lorem",
      "isDisabled": false,
      "onClick": [Function],
      "style": null,
      "to": null,
    },
    "ref": null,
    "rendered": "Lorem",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "buttonStyle": "primary",
        "children": "Lorem",
        "isDisabled": false,
        "onClick": [Function],
        "style": null,
        "to": null,
      },
      "ref": null,
      "rendered": "Lorem",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
